src/libs/ReportActionsUtils.js

```diff
@@ -371,14 +371,18 @@ function shouldReportActionBeVisibleAsLastAction(reportAction) {
  * @return {Object}
  */
 function getLastVisibleAction(reportID, actionsToMerge = {}) {
-    const actions = _.toArray(lodashMerge({}, allReportActions[reportID], actionsToMerge));
-    const visibleActions = _.filter(actions, (action) => shouldReportActionBeVisibleAsLastAction(action));
+    const actions = Object.values({
+        ...allReportActions[reportID],
+        ...actionsToMerge,
+    });
+    const visibleActions = actions.filter((action) => shouldReportActionBeVisibleAsLastAction(action));
```

Comment:
inside of `shouldReportActionBeVisibleAsLastAction` are a lot of `_.isEmpty` or underscore functions, probably we can apply the same logic you did here to improve more the performance